src/Jobs/PullStock.php

```diff
@@ -50,45 +62,65 @@ public function middleware()
         ];
     }
 
-    public function handle() {
+    public function handle(): void
+    {
         // Resolve Picqer Connection
         $connection = ConnectionFactory::getConnection();
 
-        // Filter Exact items based on Daalder product sku
-        $code = $this->product->sku;
-        $item = new Item($connection);
-        $item = $item->filter("Code eq '".$code."'");
+        if(!$this->exactItemId) {
```

Comment:
What is the benefit of filtering on the exactItemId instead of the product code? Seems like a redundant addition to this job.

Reply:
To elaborate: if there is no Exact Item that matches the `sku` of `$this->product`, something is wrong here and we should stop execution. Also, the `$this->exactItemId` will be (should be!) equal to `$this->product->exact_id` anyway.

Reply:
This exactItemId was added so that the job dispatched from the webhook controller with the new changes doesn't have to get the item again.

Reply:
@anis790615 Makes sense, but wouldn't it be cleaner to just use `$this->product->exact_id` then? You can keep the Item fetching as a fallback, but that should cover most cases.

Reply:
@vincentvanhoven  Where would you get the exact_id from? It's not saved on the product and there is not pivot table for it.

Reply:
@anis790615 You can find more info in the method ProductRepository::getProductFromExactId

Reply:
@vincentvanhoven The method is literally the same fetch from exact 
```
$connection = ConnectionFactory::getConnection();
        if($connection->needsAuthentication()) {
            throw new \Exception("Connection to Exact is not authenticated! Call the 'authenticate-exact' endpoint to fix this.");
        }

        $item = new Item($connection);
        $item = $item->filter("ID eq guid'$exactId'");

        if(count($item) === 0) {
            return null;
        }

        $item = $item[0];
        $product = Product::firstWhere('sku', $item->Code);
        if(is_null($product)) {
            return null;
        }

        return $product;
```

Reply:
Well, that's very weird. I recall it saving the exact_id somewhere on the Product 🤔 
I do believe I'm being too nitpicky on this one, so you can go ahead and merge this PR